pkg/cloud/services/securitygroup/securitygroups.go

```diff
@@ -160,6 +160,10 @@ func (s *Service) ReconcileSecurityGroups() error {
 			// skip rule reconciliation, as we expect the in-cluster cloud integration to manage them
 			continue
 		}
+		if sg.Name == "default" {
+			// skip rule reconciliation, as default SG group is already handled by revokeIngressAndEgressRulesFromVPCDefaultSecurityGroup
+			continue
+		}
```

Comment:
This special handling could lead to more bugs. For example, `authorizeSecurityGroupIngressRules` is still below and is also exempted with this `continue` statement.

Can we instead rather fix this by ignoring the `NotFound` error where it happens?

(We're also observing this annoying problem for non-EKS clusters, by the way.)